llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -145,25 +145,27 @@ bool llvm::canAdvanceOp(MachineInstr &MemI, MachineInstr &Dest,
 
 /// Find the def instruction for \p Reg, folding away any trivial copies and
 /// bitcasts. May return nullptr if \p Reg is not a generic virtual register.
+/// The \p AllowMultiUse flag permits folding even if the def instruction for \p
+/// Reg, has multiple uses.
 MachineInstr *
-llvm::getDefIgnoringCopiesAndBitcasts(Register Reg,
+llvm::getDefIgnoringCopiesAndBitcasts(Register Reg, bool AllowMultiUse,
                                       const MachineRegisterInfo &MRI) {
 
```

Comment:
You seem to never use `AllowMultiUse=true`. Do you have future plans outside of this PR?

Reply:
Oh okay there is one use in `matchExtractConcat`. I would be curious to see if allowing multiple uses in other combiners would help as well?